task/tekton-operator-install/0.1/tekton-operator-install.yaml

```diff
@@ -0,0 +1,79 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: tekton-operator-install
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: pipelines, operator, tekton
+    tekton.dev/displayName: "Tekton Operator Install"
+spec:
+  description: >-
+    This task can be used to install Tekton pipelines and also it's 
+    components using Tekton Operator on a new cluster.
+  workspaces:
+    - name: kubeconfig
+      description: >-
+        The workspace consisting of kubeconfig file so that operator
+        can install tekton components on fresh cluster
+  params:
+    - name: TRIGGERS_VERSION
+      type: string
+      description: The version of the triggers to be installed
+      default: ""
+    - name: DASHBOARD_VERSION
+      type: string
+      description: The version of the Tekton dashboard to be installed
+      default: ""
+    - name: EXTENSION_WEBHOOKS_VERSION
+      type: string
+      description: >-
+        The version of the Tekton Extension Webhooks to be installed
+      default: ""
+  steps:
+    - name: intall-components
```

Comment:
typo, `install`

shall we split this step into 2 `setup` and `install`

in setup we can write the manifests in to a dir after checking the params

in install we can call `kubectl apply -f <path to dir> ` after installing the operator

Reply:
can we avoid writing two steps because I am taking care of the following issue https://github.com/tektoncd/catalog/issues/441. I can understand your point that it will make the code more cleaner :sweat_smile:

Reply:
i see, cool :+1: 

then could you move 
```
kubectl apply -f https://storage.googleapis.com/tekton-releases/operator/latest/release.notags.yaml
```
to the top (right after KUBECONFIG check)

It would be simpler if we move the CR creating part into a function. The function will need only addon name as argument. We can write the manifest to a known path (`addon-manifest-dir`)

this would help us check params, call the function to generate CR manifests. and finally call a `kubectl apply -f` on the addon-manifest-dir.

Reply:
> can we avoid writing two steps because I am taking care of the following issue #441. I can understand your point that it will make the code more cleaner sweat_smile

I actually think it's a good fit for 2 steps :upside_down_face: one that deploy the thing, one that waits for it to be correctly installed. Steps are relatively cheap (especially if using the same image).